src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -941,6 +942,7 @@ private BufferedImage getImageRequest(String path) throws IOException, Interrupt
     private CloseableHttpClient getHttpClient(final HttpRequestBase request) {
         HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
         httpClientBuilder.useSystemProperties();
+        httpClientBuilder.setRetryHandler(new StandardHttpRequestRetryHandler());
```

Comment:
Should this set `StandardHttpRequestRetryHandler` with some values like `StandardHttpRequestRetryHandler(3, true)` based on the docs?

https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/impl/client/StandardHttpRequestRetryHandler.html#StandardHttpRequestRetryHandler(int,%20boolean)

Reply:
Another comment is that this retry handler does not retry `POST` requests.

Reply:
Thank you for your careful review! Honestly I just copied from #214.

Looking at the source now, `StandardHttpRequestRetryHandler()` is [`StandardHttpRequestRetryHandler(3, false)`](https://github.com/zzphamvanthanhzz/httpclient-4.5.2-java/blob/7b7123f5c7c36138d41353ba5497dc71dd0d4324/httpclient/src/main/java/org/apache/http/impl/client/StandardHttpRequestRetryHandler.java#L70). So, as you said, it won't retry POST requests, because they aren't considered idempotent.

But I'd argue that _not_ repeating POST requests is the safe thing to do here, otherwise it could lead to duplicated comments, etc. Regarding my own use-case, we haven't configured anything that would require a POST. #214 even explicitly said _idempotent requests_ in the title, so I changed this PR's title too. 

So, if you agree, I'd like to leave it as it is, or change it to `StandardHttpRequestRetryHandler(3, false)` if you feel that's better (not sure about best practice here, Java isn't really my forte)